base/src/main/java/org/wildfly/security/WildFlyElytronBaseProvider.java

```diff
@@ -189,14 +191,43 @@ protected void putAlgorithmParametersImplementations() {
 
     protected class ProviderService extends Service {
 
+        private final boolean withProvider;
+        private final boolean reUsable;
         private volatile Reference<Class<?>> implementationClassRef;
+        private volatile Reference<Object> instance;
 
         public ProviderService(Provider provider, String type, String algorithm, String className, List<String> aliases, Map<String,String> attributes) {
+            this(provider, type, algorithm, className, aliases, attributes, true, false);
+        }
+
+        public ProviderService(Provider provider, String type, String algorithm, String className, List<String> aliases, Map<String,String> attributes, boolean withProvider,  boolean reUsable) {
             super(provider, type, algorithm, className, aliases, attributes);
+            this.withProvider = withProvider;
```

Comment:
@darranl based on what is decided withProvider value? can it happen that a provider will be available on classpath even though this value was hard coded as false?